gtk4/src/builder_rust_scope/imp.rs

```diff
@@ -0,0 +1,122 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::prelude::*;
+use crate::subclass::prelude::*;
+use crate::Builder;
+use crate::BuilderClosureFlags;
+use crate::BuilderError;
+use crate::BuilderScope;
+use glib::translate::*;
+use glib::Closure;
+use glib::RustClosure;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+type Callback = dyn Fn(&[glib::Value]) -> Option<glib::Value>;
+
+#[derive(Default)]
+pub struct BuilderRustScope {
+    pub callbacks: RefCell<HashMap<String, Rc<Callback>>>,
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for BuilderRustScope {
+    const NAME: &'static str = "GtkBuilderRustScope";
+    type Type = super::BuilderRustScope;
+    type ParentType = glib::Object;
+    type Interfaces = (BuilderScope,);
```

Comment:
```suggestion
impl ObjectSubclass for BuilderRustScope {
    const NAME: &'static str = "GtkBuilderRustScope";
    type Type = super::BuilderRustScope;
    type Interfaces = (BuilderScope,);
```
it's a default nowadays